packages/discovery-provider/integration_tests/tasks/test_index_user_bank.py

```diff
@@ -667,42 +707,197 @@ def test_process_user_bank_txs_details_ignore_payment_router_transfers(app):
+
+    challenge_event_bus = create_autospec(ChallengeEventBus)
+
+    populate_mock_db(db, test_entries)
+
+    with db.scoped_session() as session:
+        process_user_bank_tx_details(
+            solana_client_manager=solana_client_manager_mock,
+            session=session,
+            redis=redis,
+            tx_info=tx_response,
+            tx_sig=tx_sig_str,
+            timestamp=datetime.now(),
+            challenge_event_bus=challenge_event_bus,
+        )
+
+        tip = session.query(UserTip).filter(UserTip.signature == tx_sig_str).first()
+        assert tip is not None
+        assert tip.sender_user_id == 3
+        assert tip.receiver_user_id == 4
+        assert tip.amount == 10000000000
+
+        sender_transaction_record = (
+            session.query(AudioTransactionsHistory)
+            .filter(AudioTransactionsHistory.signature == tx_sig_str)
+            .filter(AudioTransactionsHistory.user_bank == sender_waudio_user_bank)
+            .first()
+        )
+        assert sender_transaction_record is not None
+        assert sender_transaction_record.transaction_type == TransactionType.tip
+        assert sender_transaction_record.method == TransactionMethod.send
+        assert sender_transaction_record.change == -10000000000
+
+        receiver_transaction_record = (
+            session.query(AudioTransactionsHistory)
+            .filter(AudioTransactionsHistory.signature == tx_sig_str)
+            .filter(AudioTransactionsHistory.user_bank == receiver_waudio_user_bank)
+            .first()
+        )
+        assert receiver_transaction_record is not None
+        assert receiver_transaction_record.transaction_type == TransactionType.tip
+        assert receiver_transaction_record.method == TransactionMethod.receive
+        assert receiver_transaction_record.change == 10000000000
 
 
 # Transfer of WAUDIO to a non-userbank (external transfer)
 def test_process_user_bank_txs_details_transfer_audio_external_tx(app):
-    # TODO
-    return
+    tx_response = mock_valid_waudio_transfer_from_user_bank_to_external_address
+    with app.app_context():
+        db = get_db()
+        redis = get_redis()
+
+    solana_client_manager_mock = create_autospec(SolanaClientManager)
+
+    transaction = tx_response.value.transaction.transaction
+
+    tx_sig_str = str(transaction.signatures[0])
+
+    challenge_event_bus = create_autospec(ChallengeEventBus)
+
+    populate_mock_db(db, test_entries)
+
+    with db.scoped_session() as session:
+        process_user_bank_tx_details(
+            solana_client_manager=solana_client_manager_mock,
+            session=session,
+            redis=redis,
+            tx_info=tx_response,
+            tx_sig=tx_sig_str,
+            timestamp=datetime.now(),
+            challenge_event_bus=challenge_event_bus,
+        )
+
+        tip = session.query(UserTip).filter(UserTip.signature == tx_sig_str).first()
+        assert tip is None
```

Comment:
Should we be validating the value of the tip and the sender/receiver IDs?

Reply:
this is an external transfer and making sure that there is no tip. the other tip test checks sender & receiver!